src/praisonai/praisonai/code/tools/execute_command.py

```
@@ -348,16 +348,46 @@ def run_python(
-    escaped_code = code.replace('\\', '\\\\').replace('"', '\\"')
-    command = f'{python_cmd} -c "{escaped_code}"'
-    
-    return execute_command(
-        command=command,
-        cwd=cwd,
-        timeout=timeout,
-    )
+    work_dir = cwd or os.getcwd()
+
+    try:
+        result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)